### PR TITLE
DYN-7111 Enable package search to show results for package authors as the search key

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

The PR adds package authors to indexed data, this will enable users to search for packages based on the username of the desired author.

Before - After screenshots for search comparison:

d5
d4
d3
d2
d1

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

- Enable users to search for packages based on the username of the desired author.

### Reviewers

@DynamoDS/dynamo 

### FYIs

@Amoursol

### CODE REVIEW

**Comment by zeusongit** [comment]
Unrelated test failures

File: src/DynamoCore/Utilities/LuceneSearchUtility.cs [hunk 1]
```diff
@@ -314,12 +316,13 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm)
             var booleanQuery = new BooleanQuery();
             string searchTerm = QueryParser.Escape(SearchTerm);
 
-            if (searchTerm.Contains('.'))
-                searchType = SearchType.ByCategory;
-            else if (searchTerm.Contains(' '))
-                hasEmptySpaces = true;
-            else
-                searchType = SearchType.Normal;
+            if (!IsPackageContext)
```
Comment (QilongTang) [inline src/DynamoCore/Utilities/LuceneSearchUtility.cs:319]: For packages the `.` would not work right?

**Review by QilongTang — COMMENTED** [review]
LGTM, one question mainly about the function override

File: src/DynamoCore/Utilities/LuceneSearchUtility.cs (hunk 1 above)
Comment (zeusongit) [reply src/DynamoCore/Utilities/LuceneSearchUtility.cs:319]: yes

Comment (zeusongit) [reply src/DynamoCore/Utilities/LuceneSearchUtility.cs:319]: Do you mean that `.` will not be searched?

**Comment by zeusongit** [comment]
Build passed here: https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/16215/

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
Any documentation we need to update after this PR?

**Comment by zeusongit** [comment]
> Any documentation we need to update after this PR? Internal wiki about Lucene?

I was wondering if we have that, since there are a lot of conditions and customizations, I don't think we have any, but maybe @RobertGlobant20 knows?

**Pull request merged**